### PR TITLE
Adding CloudWatch Publisher Plugin onto the catalog

### PR DESCRIPTION
**Pull request by iceycake**
Fixes #

Summary of changes:
- Adding CloudWatch publisher plugin onto the Plugin Catalog

Testing done:
- See unit test in the repo 

@intelsdi-x/snap-maintainers

### CODE REVIEW

**Comment by IRCody** [comment]
Thank you for submitting this @iceycake. We were thinking it would be nice to add this to a new section.  Would you mind updating your PR to include your plugin under a "Community Plugin" section with the following info:

```md
## Community Plugins
| Name  | Type  | Description | Link |
| :---- | :---- | :---------- | :--- |
| CloudWatch | Publisher | Publishes snap metrics to AWS CloudWatch | [snap-plugin-collector-cloudwatch](https://github.com/Ticketmaster/snap-plugin-publisher-cloudwatch) |
```

**Comment by iceycake** [comment]
Updated following instructions.

File: docs/PLUGIN_CATALOG.md [hunk 1]
```diff
@@ -53,6 +53,11 @@ This is the master catalog of plugins for snap. The plugins in this list may be
 | Riemann | Publisher | Writes to Riemann monitoring system | [snap-plugin-publisher-riemann](https://github.com/intelsdi-x/snap-plugin-publisher-riemann) | [Linux](https://s3-us-west-1.amazonaws.com/snap-plugins-linux-latest/snap-plugin-publisher-riemann) &#124; [Darwin](https://s3-us-west-1.amazonaws.com/snap-plugins-darwin-latest/snap-plugin-publisher-riemann)
 | Tag | Processor | Processes data and add tags | [snap-plugin-processor-tag](https://github.com/intelsdi-x/snap-plugin-processor-tag) |
 
+## Community Plugins
+| Name  | Type  | Description | Link |
+| :---- | :---- | :---------- | :--- |
+| CloudWatch | Publisher | Publishes snap metrics to AWS CloudWatch | [snap-plugin-collector-cloudwatch](https://github.com/Ticketmaster/snap-plugin-publisher-cloudwatch) |
```
Comment (IRCody) [inline docs/PLUGIN_CATALOG.md]: Looks good except I think this should be snap-plugin-publisher-cloudwatch on the link? I just noticed my earlier comment also had that error.

**Comment by iceycake** [comment]
@IRCody Fixed the plugin name typo

**Comment by danielscottt** [comment]
LGTM

**Pull request merged**